src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -94,9 +94,17 @@ protected virtual string GetResourcePrefix(TypeInfo typeInfo, string baseNamespa
                 throw new ArgumentNullException(nameof(baseNamespace));
             }
 
-            return string.IsNullOrEmpty(resourcesRelativePath)
-                ? typeInfo.FullName
-                : baseNamespace + "." + resourcesRelativePath + TrimPrefix(typeInfo.FullName, baseNamespace + ".");
+            if (string.IsNullOrEmpty(resourcesRelativePath))
+            {
+                return typeInfo.FullName;
+            }
+            else
+            {
+                // This expectation is defined by dotnet's automatic resource storage.
+                // We have to conform to "{RootNamespace}.{ResourceLocation}.{FullTypeName - AssemblyName}".
+                var assemblyName = new AssemblyName(typeInfo.Assembly.FullName).Name;
+                return baseNamespace + "." + resourcesRelativePath + TrimPrefix(typeInfo.FullName, assemblyName + ".");
```

Comment:
Trim the `baseNamespace`, `resourcesRelativePath` for `.` here and any other methods that users(as they are open for extension) can passing in value.

Reply:
Good point that we need to start doing that, but we should do so holistically for all such inputs at once and add tests, so I've created aspnet/Home#2792 since it seems like a larger item.